tests/distributions/test_spanning_tree.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import absolute_import, division, print_function
+
+from collections import Counter
+
+import pytest
+import torch
+
+import pyro
+from pyro.distributions.spanning_tree import (NUM_SPANNING_TREES, SpanningTree, make_complete_graph, sample_tree_approx,
+                                              sample_tree_mcmc)
+from tests.common import assert_equal, xfail_if_not_implemented
+
```

Comment:
does it make sense to add any tests that check that the two backends are giving the same results? (assuming randomness can be suitably fixed)

Reply:
I'd rather test the correctness of each backend independently and avoid making assumptions about C++ vs Python rngs and data structure ordering (e.g. order of operations can be slightly different when iterating over a Python `set()` versus a C++ `std::unordered_set`).